### PR TITLE
[tracks/javascript/exercises/translation-service] Update instructions.md 

### PR DESCRIPTION
**Pull request by gsonnenf**
In this tutorial, I often couldn't tell if a problem was due to a misunderstanding of promises or due to my misunderstanding of the API .  To help with this:

* I've edited the instructions to give more detailed description of the API and remove ambiguity (so users can put their focus on learning promises).

* I've also added a bit of clean up, fixed some typos and restructured to help the flow and uniformity (It looks like there were two different authors with different writing styles and emphasis).

I hope you guys find this an improvement.

### CODE REVIEW

File: exercises/concept/translation-service/.docs/instructions.md [hunk 1]
```diff
@@ -52,10 +48,13 @@ Ensure that you *never* request a translation if something has already been tran
...
 ## 1. Fetch a translation, ignoring the quality
 
-Implement a function `free(text)` to fetch a translation, ignoring the quality, and forwarding any errors thrown by the API:
+The free service only provides translations that are currently in our _datastore_. 
+
+Implement a method `free(text)` that provides free members with translation that already exist in the _datastore_. Ignore the quality and forward any errors thrown by the API.
 
 - Returns the translation if it can be retrieved, regardless of its quality
 - Forwards any error from the translation API
+- Uses the `api.fetch` method ( remember that `api.fetch` returns a `promise` )
```
Comment (SleeplessByte) [inline exercises/concept/translation-service/.docs/instructions.md:57]:
```suggestion
- Uses the `api.fetch` method (`api.fetch` returns a `promise`)
```

File: exercises/concept/translation-service/.docs/instructions.md [hunk 2]
```diff
@@ -99,11 +98,11 @@ service.request("jIyajbe'");
 
 ## 4. Fetch a translation, inspect the quality, or request it
 
-Implement the function `premium(text, quality)` for premium users, which fetches a translation, requests it if it's not available, and only returns it if it meets a certain threshold.
+Implement a premium user method `premium(text, quality)` to `api.fetch` a translation. If a translation is `notAvailable`, `api.request` the translation and `api.fetch` it after its been added to the _datastore_. The method should only return the translation if it meets a certain `quality` threshold.
```
Comment (SleeplessByte) [inline exercises/concept/translation-service/.docs/instructions.md:101]:
```suggestion
Implement a premium user method `premium(text, quality)` to fetch a translation.
If a translation is `NotAvailable`, request the translation and fetch it after its been added to the _datastore_.
The method should only return the translation if it meets a certain `quality` threshold.
```

Again, I'm not a fan of making the instructions so context-less that you no longer need the story.

**Comment by gsonnenf** [comment]
This was a point of confusion for me when making a request. I expected the request would return the translation, not add it to storage. I had to read the API code, and the documentations a few time to understand it was implied.

Perhaps replace the datastore language with cache, local storage, database or something. I don't think I'm adding a new concept. I am just describing a concept that already exists in the API. It is helpful to understand the difference between the fetch and request functionality.

File: exercises/concept/translation-service/.docs/instructions.md [hunk 3]
```diff
@@ -1,38 +1,34 @@
...
+In this exercise, you'll be providing a `TranslationService` that provides basic translation services to free members, and advanced translation to premium members with quality assurances.
 
-You have found an outer space translation API that can fulfill any translation _request_ in a reasonable amount of time, and you want to capitalize on this.
+**The API**
 
-**The API interface**
+You have found an outer space translation API that fulfills any translation `request` in a reasonable amount of time. You want to capitalize on this. The space translators are extremely fickle and hate redundancy, so they also provide a _datastore_ in their API where you can `fetch` past translations without bothering them.
 
-The API has a very minimal interface:
+***Fetching a translation***
```
[the remark this answers is not in the document]
Comment (gsonnenf) [reply exercises/concept/translation-service/.docs/instructions.md:9]: image

File: exercises/concept/translation-service/.docs/instructions.md [hunk 4]
```diff
@@ -1,38 +1,34 @@
...
-An `Untranslatable` error is thrown if a piece of text is untranslatable.
+If a translation is not found in the _datastore_, the API throws a `NotAvailable` error( Translations can be added using the `api.request` method ). If 'text' is not translatable, the API throws a `Untranslatable` error. 
 
 ```javascript
 api.fetch('jIyaj');
 // => Promise({ resolved: 'I understand' })
 ```
 
-**Requesting a translation**
+***Requesting a translation***
```
[the remark this answers is not in the document]
Comment (gsonnenf) [reply exercises/concept/translation-service/.docs/instructions.md:25]: Italics and bold to differentiate from the bold header

File: exercises/concept/translation-service/.docs/instructions.md [hunk 5]
```diff
@@ -1,38 +1,34 @@
...
+
+Some translations are sure to exist, but haven't been added to our _datastore_ yet. That's the difference between `NotAvailable` ( not in our datastore, but can be requested ) and `Untranslatable` ( cannot be translated ).
```
[the remark this answers is not in the document]
Comment (gsonnenf) [reply exercises/concept/translation-service/.docs/instructions.md:27]: Probably better to keep these as one block as the first item references the second item. They are not discrete ideas.

File: exercises/concept/translation-service/.docs/instructions.md (hunk 2 above)
Comment (gsonnenf) [reply exercises/concept/translation-service/.docs/instructions.md:112]: I think its easier for to learn promises, a tricky subject, if your explicit about the example API [if a side goal is not learning how to deal with sparsely documented API]. 

I think the changes you made give enough clarity that it won't give people API trouble.

File: exercises/concept/translation-service/.docs/instructions.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (gsonnenf) [reply exercises/concept/translation-service/.docs/instructions.md:11]: Oh, also this sub-commit managed to confuse changing the formatting with removal of a less useful line. Might want to remove that again, if italic/bold is changed.

[the remark this answers is not in the document]
Comment (SleeplessByte) [reply exercises/concept/translation-service/.docs/instructions.md:11]: Right. I find that personally very confusing. I usually do `**_something_**`, but it's likely `prettier` formats that as `***`. In that case, leave as is :)

File: exercises/concept/translation-service/.docs/instructions.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply exercises/concept/translation-service/.docs/instructions.md:29]: A return in markdown doesn't create a new block. Two returns do.

**Comment by SleeplessByte** [comment]
> This was a point of confusion for me when making a request. I expected the request would return the translation, not add it to storage. I had to read the API code, and the documentations a few time to understand it was implied.

Ah. That comment has clarified what you tried to achieve with `datastore`. Perhaps indeed `cache`, even though technically incorrect, may be clear for more people? Like "requesting a translation adds it to the service's cache"?. I'm okay with `datastore` too, it's just not a word that we can expect people to understand. Cache isn't much better, by the way. Perhaps it makes sense to write it out instead. "When a translation is requested, it is also ..."

**Comment by SleeplessByte** [comment]
(Sidenote: next time, if you go to the review tab, you can batch apply suggestions!)

File: exercises/concept/translation-service/.docs/instructions.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (gsonnenf) [reply exercises/concept/translation-service/.docs/instructions.md:29]: I meant a colloquial definition of block, as in part of the same paragraph without a line break. No matter, either way isn't an issue.

**Comment by gsonnenf** [comment]
Perhaps call it storage or even a database? I'm sure people will understand what those mean on a basic level: "A place where data is stored". 

In the same way we pretend space translators exist, we can pretend a database exists in the API.

**Comment by gsonnenf** [comment]
Perhaps call it storage or even a database? I'm sure people will understand what those mean on a basic level: "A place where data is stored". In the same way we pretend space translators exist, we can pretend a database exists in the API.

(TY for tip, I don't do alot of Github pull requests)

**Comment by SleeplessByte** [comment]
Sounds good. want to make the update to storage? 

@junedev would you be so kind to spot check as well?

**Review by gsonnenf — COMMENTED** [review]
Changes made.

**Comment by gsonnenf** [comment]
Can someone who is familiar with this repo resolve the prettier issue?

**Review by SleeplessByte — APPROVED** [review]

**Comment by SleeplessByte** [comment]
Thank you for these changes and additions @gsonnenf . I think it makes the exercise much better.

**Pull request merged**